tests/python_on_whales/components/test_system.py

```diff
@@ -35,3 +36,76 @@ def test_parsing_events():
     for event in events:
         parsed: DockerEvent = DockerEvent.parse_obj(event)
         assert parsed.time.date() == date(2020, 12, 28)
+
+
+def test_prune_prunes_image():
+    # TODO: Test dangling image
+    for container in docker.container.list(filters={"ancestor": "busybox"}):
+        docker.container.remove(container, force=True)
+    image = docker.pull("busybox")
+    assert image in docker.image.list()
+
+    # image not pruned because not dangling
+    docker.system.prune()
+    assert image in docker.image.list()
+
+    # image not pruned because it does not have dne label
+    docker.system.prune(all=True, filters={"label": "dne"})
+    assert image in docker.image.list()
+
+    # image pruned
+    docker.system.prune(all=True)
+    assert image not in docker.image.list()
+
+
+def test_prune_prunes_container():
+    stopped_container = docker.run("hello-world", remove=False, detach=True)
+    running_container = docker.run(
+        "ubuntu", ["sleep", "infinity"], remove=False, detach=True
+    )
+
+    assert stopped_container in docker.container.list(all=True)
+    assert running_container in docker.container.list()
+
+    docker.system.prune()
+
+    assert stopped_container not in docker.container.list(all=True)
+    assert running_container in docker.container.list()
+    docker.container.remove(running_container, force=True)
+
+
+def test_prune_prunes_network():
+    try:
+        with docker.network.create(random_name()) as my_net:
+            assert my_net in docker.network.list()
+            docker.system.prune()
+            assert my_net not in docker.network.list()
+    except DockerException:

```

Comment:
I believe too many pieces of code can throw a DockerException. It might hide a real bug in the test. Can you try to make the `try:... Except:` narrower? Or you could create the network with a random name, meaning that it won't have any effect if, by chance, the prune doesn't remove it.

Reply:
Good point about this being too broad.

I don't quite understand what you are talking about with the random name. I create the network with _random_name_. 

I'll think of a way to make this work.

Reply:
Sorry, I wasn't clear. What I meant was that since the network has a random name, you don't need to delete it, meaning that you don't need to use `with`, meaning that you don't need a `try except`

Reply:
I was deleting it -- twice! Point taken that I don't need to worry too much about a network with a randomly generated name floating around.

I got rid of the context manager. I don't need to worry about `__exit__` not being called because cleaning the network is exactly what this test tests.